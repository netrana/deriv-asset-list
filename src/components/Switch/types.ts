import { SwitchProps } from 'antd';

export interface Props extends SwitchProps {}
