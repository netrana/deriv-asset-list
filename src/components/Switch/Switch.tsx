import React, { FC } from 'react';

import { Switch as AntdSwitch, Typography } from 'antd';
import classNames from 'classnames';

const { Text } = Typography;

import { Props } from './types';
import styles from './switch.module.scss';

const Switch: FC<Props> = (props) => {
  const { title, className = '', ...otherProps } = props;

  return (
    <div
      className={classNames({
        [styles.switch]: true,
        [className]: Boolean(classNames),
      })}
    >
      <AntdSwitch {...otherProps} />
      {!!title ? (
        <Text className={styles.title} strong>
          {title}
        </Text>
      ) : null}
    </div>
  );
};

export default Switch;
