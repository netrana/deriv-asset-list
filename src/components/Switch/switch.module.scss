@import 'styles/globals.scss';

.switch {
  display: flex;
  align-items: center;

  :global(.ant-switch-checked) {
    background-color: $primaryThemeColor !important;
    border-color: $primaryThemeColor;
  }

  .title {
    margin-left: 16px;
    font-family: Poppins;
    font-weight: 500;
    font-size: 18px;
    line-height: 27px;
    color: rgba(0, 0, 0, 0.87);
  }
}
