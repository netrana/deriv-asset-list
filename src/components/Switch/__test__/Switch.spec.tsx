import React from "react";
import render from "react-test-renderer";

import Switch from "../Switch";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Switch: "Switch",
  Typography: {
    ...jest.requireActual("antd").Typography,
    Text: "Text",
  },
}));

describe("Switch component test", () => {
  describe("snapshot test", () => {
    it("should match snapshot without title", () => {
      const component = render.create(<Switch />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    it("should match snapshot with title", () => {
      const component = render.create(<Switch title="Test Title" />);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
