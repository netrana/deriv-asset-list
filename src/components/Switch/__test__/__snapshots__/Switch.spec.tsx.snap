// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component test snapshot test should match snapshot with title 1`] = `
<div
  className="switch "
>
  <Switch />
  <Text
    className="title"
    strong={true}
  >
    Test Title
  </Text>
</div>
`;

exports[`Switch component test snapshot test should match snapshot without title 1`] = `
<div
  className="switch "
>
  <Switch />
</div>
`;
